activesupport/lib/active_support/message_verifier.rb

```diff
@@ -106,12 +115,26 @@ def verify(signed_message)
     #
     #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
     #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
-    def generate(value)
-      data = encode(@serializer.dump(value))
-      "#{data}--#{generate_digest(data)}"
+
+    def generate(options)
```

Comment:
`generate` is public API, so we can't do that without a deprecation cycle.

I think `value` is the worst variable name possible, so I'd find words that communicate more where possible :wink:
`generate(value: 'abc')` doesn't add much over `generate('abc')`.

You're letting the `value` variables in the cookies code seep in to here, which I think is the wrong direction to go. Lets keep those things separate. Remember, people can use the verifier and encryptor outside of cookies.